sycl/source/detail/graph_impl.hpp

```diff
@@ -434,6 +381,9 @@ class graph_impl {
   /// List of root nodes.
   std::set<std::shared_ptr<node_impl>> MRoots;
 
+  /// Storage for all nodes contained within a graph
```

Comment:
If there's anything you can elaborate on in this comment about how this vector is used it would be handy.

Reply:
Adding explanation of usage here.